packages/scripts/src/migration/dvm2-designated-voting/0_Deploy.ts

```diff
@@ -1,111 +1,68 @@
 
+async function main() {
   const networkId = Number(await hre.getChainId());
+  console.log("Running DesignatedVotingV2 deployments🔥\nChainId:", networkId);
+
   if (networkId != 1) throw new Error("Can only run on mainnet");
-  if (!process.env.ETHERSCAN_API_KEY) throw new Error("No ETHERSCAN_API_KEY");
   if (!process.env.OWNER_TO_MIGRATE) throw new Error("No OWNER_TO_MIGRATE set");
   const owner = process.env.OWNER_TO_MIGRATE || "";
+  const maxDesignatedVotingDeployedPerBatch = Number(process.env.MAX_CONTRACTS_PER_BATCH) || 5;
 
-  const factoryV1 = await getContractInstance<DesignatedVotingFactoryEthers>("DesignatedVotingFactory");
-  // const factoryV2 = await getContractInstance<DesignatedVotingV2FactoryEthers>("DesignatedVotingV2Factory");
-  const votingToken = await getContractInstance<VotingTokenEthers>("VotingToken");
-  // Step 1: find all DesignatedVoting contracts created by the DesignatedVotingFactory.
-  const designatedVotingContracts = await _fetchDesignatedVotingContractsCreatedByFactory(factoryV1);
-
-  // Step 2: remove all DesignatedVoting contracts that have 0 UMA.
-  const contractsTokenBalances = await Promise.all(
-    designatedVotingContracts.map(async (address) => votingToken.balanceOf(address))
-  );
-  const designatedVotingContractsWithTokens: DesignatedVotingEthers[] = [];
-  for (let i = 0; i < designatedVotingContracts.length; i++)
-    if (contractsTokenBalances[i].gt(0))
-      designatedVotingContractsWithTokens.push(
-        await getContractInstance<DesignatedVotingEthers>("DesignatedVoting", designatedVotingContracts[i])
-      );
-
-  // Step 3: Find all owners and hot wallets for all DesignatedVoting contracts that have UMA. Construct one condensed
-  // data structure that contains all key information. Remove any that are not owned by the provided owner.
-
-  const designatedVotingData = (
-    await Promise.all(
-      designatedVotingContractsWithTokens.map(async (contract) => {
-        const contractOwner = await contract.getMember(0);
-        const hotWallet = await contract.getMember(1);
-        // Typescript knows these values are all defined.
-        return { designatedVoting: contract.address, owner: contractOwner, hotWallet };
-      })
-    )
-  ).filter((contract) => contract.owner === owner);
+  // Fetch all current DesignatedVoting contracts owned by the owner. Remove elements that have 0 balance.
+  const designatedVotingData = (await getDesignatedVotingContractsOwnedByOwner(owner)).filter((e) => e.balance.gt(0));
 
   // Log all designated voting and the associated hot wallets. remove the owner element from the object to keep it short.
   console.log(`Found the following DesignatedVoting to migrate owned by ${owner}:`);
   console.table(
-    designatedVotingData.map((e: { designatedVoting: string; owner?: string; hotWallet: string }) => {
+    designatedVotingData.map((e: { designatedVoting: string; owner?: string; voter: string; balance: any }) => {
```

Comment:
Shouldn't this follow this type? Why are some of the values in here different?

export interface DesignatedVotingOwnerWithBalance {
  designatedVoting: string;
  owner: string;
  voter: string;
  balance: BigNumber;
}

Reply:
this is just for logging; I need to preform some changes on the structure of the data to make it nicely human readable and such. I've made it consistent with the other file.